README.md

```diff
@@ -1,76 +1,122 @@
+The program brings up an underlying network interface and if it's Wifi also scans for access points.
+It creates a TCPSocket and performs an HTTP transaction targeting the website in the `mbed_app.json` config.
 
-The program brings up an underlying network interface, and uses it to perform an HTTP
-transaction over a TCPSocket.
+(Note: To see this example in a rendered form you can import into the Arm Mbed Online Compiler,
+please see [the documentation](https://os.mbed.com/docs/mbed-os/latest/apis/socket.html#socket-example).)
 
-### Selecting the network interface
+## Selecting the network interface
 
-This application is able to use any network inteface it finds. Please see the Mbed OS documentationg for [selecting the default network interface](https://os.mbed.com/docs/v5.10/apis/network-interfaces.html).
+This application is able to use any network interface it finds.
+Please see the Mbed OS documentation for [selecting the default network interface](https://os.mbed.com/docs/v5.10/apis/network-interfaces.html).
 
-For example, building on Ethernet enabled boards, you do not do any configuration.
+For example, building on Ethernet enabled boards, you do not need any further configuration.
 
-Building for WiFi boards, you need to provide SSID, password and security settings in `mbed_app.json` as instructed in the documentation. For example, like this:
+### WiFi
+
+If you want to use WiFi you need to provide SSID, password and security settings in `mbed_app.json`.
+
+If your board doesn't provide WiFi as the default interface because it has multiple interfaces you need to specify that you want WiFi in `mbed_app.json`.
 
 ```
 {
     "target_overrides": {
         "*": {
-            "platform.stdio-convert-newlines": true,
             "target.network-default-interface-type": "WIFI",
-            "nsapi.default-wifi-security": "WPA_WPA2",
-            "nsapi.default-wifi-ssid": "\"ssid\"",
-            "nsapi.default-wifi-password": "\"password\""
         }
     }
 }
 ```
 
-Building for boards that have more that one network interface, you might need to override `target.network-default-interface-type` variable.
+For more information about Wi-Fi APIs, please visit the [Mbed OS Wi-Fi](https://os.mbed.com/docs/latest/reference/wi-fi.html) documentation.
+
+### Supported WiFi hardware
+
+* All Mbed OS boards with build-in Wi-Fi module:
+    * [ST DISCO IOT board](https://os.mbed.com/platforms/ST-Discovery-L475E-IOT01A/) with integrated [ISM43362 WiFi Inventek module](https://github.com/ARMmbed/wifi-ism43362).
+    * [ST DISCO_F413ZH board](https://os.mbed.com/platforms/ST-Discovery-F413H/) with integrated [ISM43362 WiFi Inventek module](https://github.com/ARMmbed/wifi-ism43362).
+* Boards with external WiFi shields.
+    * [NUCLEO-F429ZI](https://os.mbed.com/platforms/ST-Nucleo-F429ZI/) with ESP8266-01 module using pins D1 and D0.
```

Comment:
I do not understand this list, external WiFi shields can be used with most boards no ? 
Do you think we should provide a configuration with NUCLEO-F429ZI + ESP8266 ? 
What does "using pins D1 and D0" means ?

Reply:
I think this is trying to clarify what boards have been tested and are known to work fine.
Some boards may have some incompatibilities. 
@ARMmbed/team-st-mcd  let us know what you think.

Reply:
the D1 D0 comment is unclear, looks like it's already the default rx/tx pins so it's not even something you have to override, I'm removing it